components/eamxx/src/share/util/eamxx_time_interpolation.cpp

```diff
@@ -268,6 +266,19 @@ void TimeInterpolation::set_file_data_triplets(const vos_type& list_of_files) {
     } else {
       EKAT_ERROR_MSG("Error!! TimeInterpolation::set_file_triplets - unsupported units of time = (" << time_units << ") in source data file " << filename << ", supported units are: seconds, minutes, hours and days");
     }
+    auto date_start = time_units.find_first_of("0123456789");
```

Comment:
Perhaps, if you want to _assume_ that the units are "XYZ since YYYY-MM-DD HH:MM:SS", you should check that the string has indeed that format. You should use a regex here. Since you found out what the units are, you can do
```
#include <regex>
...
std::regex r { units + " since [0-9]{4,}-[0-9]{2,}-[0-9]{2,} [0-9]{2,}:[0-9]{2,}:[0-9]{2,}"};
EKAT_REQUIRE_MSG (std::regex_search(time_units,r),
  "Error! Time units should be of the form '<untis> since YYYY-MM-DD HH:MM:SS'\n"
  "  - time units: " + time_units + "\n");
```